### ISSUE TITLE
Lazy Octopus | Faster, better, cheaper wallet deployment  in less code

### ISSUE THREAD

**Issue by arielman87** [0xsequence/public-roadmap#9]
### TL;DR:
Sequence wallet is a smart contract that is deployed on the blockchain and which can be controlled by multiple private keys. Lazy Octopus will delay the on-chain deployment while giving the user a faster login, and pre-signing transaction without extra cost.

1. Rollup style signatures: Pre-sign all session openings in order to add new devices without additional fees

2. Offchain logins/logouts: Defers wallet updates until a transaction is sent

### Official Channels:
**Twitter**: https://twitter.com/0xsequence 
**Discord**: https://discord.gg/sequence 
**Reddit**: https://reddit.com/r/0xSequence/
